rules-reviewed/eap7/eap6/hibernate4.windup.xml

```diff
@@ -604,5 +572,160 @@
                 <matches pattern="(TableGenerator|TableHiLoGenerator|SequenceGenerator|SequenceIdentityGenerator|SequenceHiLoGenerator)" />
             </where>
         </rule>
+        <rule id="hibernate4-00029">
+            <when>
+                <javaclass references="org.hibernate.type.descriptor.sql.SqlTypeDescriptor">
```

Comment:
I guess that it only works transitively if the entire inheritance transitive closure is available for Windup to analyze. When migrating an EAP application, I don't expect Hibernate core to be available. But maybe I'm misunderstanding here.